neqo-transport/src/connection/mod.rs

```diff
@@ -1828,13 +1825,24 @@ impl Connection {
                     let read = self.crypto.streams.read_to_end(space, &mut buf);
                     qdebug!("Read {} bytes", read);
                     self.handshake(now, space, Some(&buf))?;
+                    let mut tokens = Vec::new();
+                    if let Agent::Client(ref mut c) = self.crypto.tls {
+                        while let Some(t) = c.resumption_token() {
+                            tokens.push(t);
+                        }
+                    }
+                    while let Some(t) = tokens.pop() {
+                        self.last_token = Some(t);
+                        self.create_resumption_token();
+                    }
```

Comment:
Let's factor this out somehow.  Probably best to make a new function in `crypto.rs` for getting any new tokens that have arrived.  That can deal with the `crypto.tls` internals.